### PR TITLE
[Torch] Add support for Float8E8M0FNU type

### PR DESCRIPTION
**Pull request by Tessil**
Add support for the microscaling Float8E8M0FNU type.

### CODE REVIEW

File: lib/Dialect/Torch/Utils/Utils.cpp [hunk 1]
```diff
@@ -342,7 +346,7 @@ Value Torch::getConstantWithGivenDtypeAndValue(PatternRewriter &rewriter,
                                  rewriter.getI64IntegerAttr((int64_t)value));
   if (dtype.isF64() || dtype.isF32() || dtype.isF16() || dtype.isBF16() ||
       isa<Float8E5M2Type, Float8E4M3FNType, Float8E5M2FNUZType,
-          Float8E4M3FNUZType>(dtype))
+          Float8E4M3FNUZType, Float8E8M0FNUType>(dtype))
```
Comment (Lallapallooza) [inline lib/Dialect/Torch/Utils/Utils.cpp]: `DecomposeAten_ToCopyOp` looks incorrect for `Float8E8M0FNU`.

`_to_copy` can lower through `aten.full_like(..., 0.0, dtype=float8_e8m0fnu)`, but the concrete runtime analogue fails locally: `x.to(dtype=torch.float8_e8m0fnu)` succeeds while `torch.full_like(x, 0.0, dtype=torch.float8_e8m0fnu)` raises `value cannot be converted to type c10::Float8_e8m0fnu without overflow`.

Please special-case `_to_copy` for this dtype, or keep `Float8E8M0FNUType` out of the generic zero-constant path until this lowering is valid and covered by a regression test.

File: python/torch_mlir/extras/fx_importer.py
Comment (Lallapallooza) [inline python/torch_mlir/extras/fx_importer.py:1]: Could we add some regression?

I'd expect:

- if the `_to_copy` decompose path stays, a decomposition test under `test/Dialect/Torch/decompose-complex-ops.mlir`

- regression coverage for the new `float8_e8m0fnu` literal-materialization path.

**Review by Lallapallooza — APPROVED** [review]
LGTM, few nits.

File: lib/Dialect/Torch/Utils/Utils.cpp (hunk 1 above)
Comment (Tessil) [reply lib/Dialect/Torch/Utils/Utils.cpp]: Thanks, I added a guard to make sure we don't create zero-value `Float8E8M0FNUType` in `getConstantWithGivenDtypeAndValue`.

**Review by zjgarvey — COMMENTED** [review]
To second Vitalii, please add some tests before merging. Otherwise this looks good, thanks!

File: python/torch_mlir/extras/fx_importer.py
Comment (Tessil) [reply python/torch_mlir/extras/fx_importer.py:1]: I added a LIT test with `f8E8M0FNU` that will materialise the type and also check the `getConstantWithGivenDtypeAndValue` path for non-zero values (more difficult to make an error test with zero due to the `llvm::report_fatal_error`, we could eventually change that).

**Comment by Tessil** [comment]
@zjgarvey Thanks, I added the necessary test.

**Comment by Lallapallooza** [comment]
LGTM

**Review by sjarus — APPROVED** [review]
